frontend/src/metabase/parameters/utils/parameter-values.js

```diff
@@ -29,80 +28,58 @@ export function getParameterValueFromQueryParams(
 ) {
   queryParams = queryParams || {};
 
-  const fields = getFields(parameter, metadata);
   const maybeParameterValue = queryParams[parameter.slug || parameter.id];
 
-  if (hasParameterValue(maybeParameterValue)) {
-    const parsedValue = parseParameterValueForFields(
-      maybeParameterValue,
-      fields,
-    );
+  // skip parsing "" because it indicates a forcefully unset parameter
+  if (maybeParameterValue === "") {
+    return "";
+  } else if (hasParameterValue(maybeParameterValue)) {
+    const parsedValue = parseParameterValue(maybeParameterValue, parameter);
     return normalizeParameterValueForWidget(parsedValue, parameter);
   } else {
     return parameter.default;
   }
 }
 
+export function parseParameterValue(value, parameter) {
+  if (Array.isArray(parameter.fields) && parameter.fields.length > 0) {
+    return parseParameterValueForFields(value, parameter.fields);
```

Comment:
Nitpick: would it be more readable to destructure `fields` first out of `parameter`?

Reply:
ya i spose